src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -63550,6 +63538,245 @@ public void LocalFunctions_Speculation()
             Assert.Equal("System.Object", objectSymbol2.ToTestDisplayString());
         }
 
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/70856")]
+        [InlineData("foreach (var c in y)")]
+        [InlineData("while (y.Length != 2)")]
+        [InlineData("for (int i = 0; i < y.Length; i++)")]
+        public void LocalFunction_Loop(string loopHead)
+        {
+            var source = $$"""
+                #nullable enable
+                class C
+                {
+                    void M()
+                    {
+                        var x = "a";
+
+                        f1();
+
+                        void f2(string y)
+                        {
+                            {{loopHead}}
+                            {
+                            }
+
+                            x.ToString();
+                        }
+
+                        void f1()
+                        {
+                            f2("b");
+                        }
+                    }
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void LocalFunction_UnreachableInOnePath()
+        {
+            var source = """
+                #nullable enable
+                class C
+                {
+                    void M()
+                    {
+                        var x = "a";
+
+                        f1();
+
+                        void f1()
+                        {
+                            x.ToString();
+                        }
+
+                        void f2()
+                        {
+                            f1();
+                        }
+                    }
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (15,14): warning CS8321: The local function 'f2' is declared but never used
+                //         void f2()
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "f2").WithArguments("f2").WithLocation(15, 14));
+        }
+
+        [Fact]
+        public void LocalFunction_UnreachableInOnePath_Worsening()
+        {
+            var source = """
+                #nullable enable
+                class C
+                {
+                    void M()
+                    {
+                        var x = "a";
+
+                        f1();
+
+                        void f1()
+                        {
+                            x.ToString(); // 1
```

Comment:
This numbered comment can be removed?

Reply:
I think more about these numbered comments as points of interest, not necessarily ones that have diagnostics. Or is that wrong?

Reply:
Typically, we use them to indicate diagnostics. So the test reads like we're missing an intended diagnostic. I'll let Rikki comment for a second opinion. It's a nit

Reply:
Numbered comments are for indicating positions of diagnostics when it is felt the VerifyDiagnostics baseline itself is not easily interpretable along with the test.

Reply:
> Numbered comments are for indicating positions of diagnostics when it is felt the VerifyDiagnostics baseline itself is not easily interpretable along with the test.

I interpret this as: "comments in the base-line point to the source, rather than comments in the source point to the base-line. In other words, baseline is the source of truth and it is Ok to have any comments in the source (up to the test developer to decide). And that is matching how I always looked at this issue. We never had a rule that comments in the test source must correspond to a baseline in any specific way.